kotlinx-coroutines-test/common/src/TestScope.kt

```diff
@@ -170,14 +208,56 @@ internal class TestScopeImpl(context: CoroutineContext) :
     private val uncaughtExceptions = mutableListOf<Throwable>()
     private val lock = SynchronizedObject()
 
+    override val backgroundScope: CoroutineScope =
+        CoroutineScope(coroutineContext + BackgroundWork + ReportingSupervisorJob {
+            if (it !is CancellationException) reportException(it)
```

Comment:
`reportException` may throw an exception in edge cases. Job's machinery is unlikely to be ready for that and it can lead to very weird deadlocks or inconsistent states.

I suggest doing something else, for example reporting it to global EH

Reply:
Do you have good ideas for the `context` for the `handleCoroutineException` invocation?